src/patternfly/components/Button/examples/button-states-example.hbs

```diff
@@ -1,49 +1,64 @@
 {{#> button button--modifier="pf-m-primary"}}
+  {{> button-icon button-icon--modifier="pf-c-button__icon"}}
   Primary
 {{/button}}
 {{#> button button--modifier="pf-m-primary pf-m-focus"}}
+  {{> button-icon button-icon--modifier="pf-c-button__icon"}}
```

Comment:
This button (and a lot of others) shouldn't have an icon. But the reason you're getting the `?` or `!` is because you haven't defined `button-icon--type`, which defines which font awesome icon to use.

Reply:
Thanks @mcoker, I took another look at the code and made some adjustments.